client/my-sites/checkout/checkout-thank-you/test/header.js

```diff
@@ -26,50 +27,57 @@ describe( 'CheckoutThankYouHeader', () => {
-			expect( comp.find( '.checkout-thank-you__header-heading' ).text() ).toBe( 'Loading…' );
+			render( <CheckoutThankYouHeader isDataLoaded={ false } { ...defaultProps } /> );
+			expect( screen.getByRole( 'heading', { level: 1 } ) ).toHaveTextContent( 'Loading…' );
 		} );
+
 		test( 'Should display getText()-based success message when isSimplified=false (default)', () => {
-			const comp = shallow( <CheckoutThankYouHeader isDataLoaded={ true } { ...defaultProps } /> );
-			expect( comp.find( '.checkout-thank-you__header-heading' ).text() ).toEqual(
+			render( <CheckoutThankYouHeader isDataLoaded={ true } { ...defaultProps } /> );
+			expect( screen.getByRole( 'heading', { level: 1 } ) ).toHaveTextContent(
 				'Congratulations on your purchase!'
 			);
-			expect( comp.find( '.checkout-thank-you__header-text' ).text() ).toEqual(
-				"Your site is now on the {{strong}}%(productName)s{{/strong}} plan. It's doing somersaults in excitement!"
+			expect( screen.getByRole( 'heading', { level: 2 } ) ).toHaveTextContent(
+				"Your site is now on the {{strong}}%(productName)s{{/strong}} plan. It's doing somersaults in excitement!"
 			);
 		} );
 		test( 'Should display an alternative success message when isSimplified=true', () => {
-			const comp = shallow(
+			render(
 				<CheckoutThankYouHeader isDataLoaded={ true } isSimplified={ true } { ...defaultProps } />
 			);
-			expect( comp.find( '.checkout-thank-you__header-heading' ).text() ).toEqual(
+			expect( screen.getByRole( 'heading', { level: 1 } ) ).toHaveTextContent(
 				'Congratulations on your purchase!'
 			);
-			expect( comp.find( '.checkout-thank-you__header-text' ).text() ).toEqual(
+			expect( screen.getByRole( 'heading', { level: 2 } ) ).toHaveTextContent(
 				'Your site is now on the {{strong}}%(productName)s{{/strong}} plan. Enjoy your powerful new features!'
 			);
 		} );
 		test( 'Should display a list of success messages when siteUnlaunchedBeforeUpgrade=true', () => {
-			const comp = shallow(
+			render(
 				<CheckoutThankYouHeader
 					isDataLoaded={ true }
 					isSimplified={ true }
 					siteUnlaunchedBeforeUpgrade={ true }
 					{ ...defaultProps }
 				/>
 			);
-			expect( comp.find( '.checkout-thank-you__header-heading' ).text() ).toEqual(
+			expect( screen.getByRole( 'heading', { level: 1 } ) ).toHaveTextContent(
 				'Congratulations on your purchase!'
 			);
-			expect( comp.find( '.checkout-thank-you__header-text' ).length ).toBe( 0 );
-			expect( comp.find( '.checkout-thank-you__success-message-item' ).length ).toEqual( 2 );
-			expect(
-				comp.find( '.checkout-thank-you__success-message-item:last-child div' ).text()
-			).toEqual(
-				"Your site has been launched. You can share it with the world whenever you're ready."
+			expect( screen.queryByRole( 'heading', { level: 2 } ) ).not.toBeInTheDocument();
+
+			const messages = screen.queryAllByRole( 'listitem' );
```

Comment:
Could we also use an assertion to ensure the number of list items found by the query here is correct?

Reply:
Done via https://github.com/Automattic/wp-calypso/pull/66108/commits/389a1c78df9e67d2f8e8eba2e9d86fc1704b462d